### PR TITLE
Do not show native JavaScript objects in scope

### PR DESCRIPTION
**Pull request by annagrin**
Currently debugger uses heuristics to translate JavaScript stack frames to dart, which results in  some references to JavaScript objects left in current frame, to be displayed by IDEs. IDEs, such as IntelliJ, can receive such a reference, and call getObject to retrieve the value, which is not supported for native JavaScript objects and will result in an exception.

This change removes native JavaScript objects from Dart stack, and makes the exception on non-supported getObject explicit. This is aimed as a short-term fix for the IntelliJ problem below. The long term solution will involve compiler and frontend-server changes to assist in translation of stacks and scopes from JavaScript to dart: #982 

Fixes: #965

### CODE REVIEW

File: dwds/lib/src/debugging/classes.dart [hunk 1]
```diff
@@ -59,6 +59,9 @@ class ClassHelper extends Domain {
     if (clazz != null) return clazz;
     var splitId = objectId.split('|');
     var libraryId = splitId[1];
+    if (libraryId == 'null') {
```
Comment (devoncarew) [inline dwds/lib/src/debugging/classes.dart:62]: Just confirming that we intend to compare to the string `'null'` and not the `null` value?

File: dwds/lib/src/debugging/debugger.dart [hunk 2]
```diff
@@ -338,14 +340,21 @@ class Debugger extends Domain {
...
+
     // TODO(alanknight): Can these be moved to dart_scope.dart?
     var properties = await visibleProperties(
         scopeList: scopeChain.cast<Map<String, dynamic>>().toList(),
         debugger: this,
         callFrameId: callFrameId);
     var boundVariables = await Future.wait(
         properties.map((property) async => await _boundVariable(property)));
-    boundVariables = boundVariables.where((bv) => bv != null).toList();
+    boundVariables = boundVariables
```
Comment (devoncarew) [inline dwds/lib/src/debugging/debugger.dart:355]: Perhaps explain why we're filtering here with a comment?

File: dwds/lib/src/debugging/classes.dart (hunk 1 above)
Comment (annagrin) [reply dwds/lib/src/debugging/classes.dart:62]: Yes, the object id has the string value  `"classes|null|NativeJavaScriptObject"` and libraryId is that middle part `"null"`, a string as well.

File: dwds/lib/src/debugging/debugger.dart (hunk 2 above)
Comment (annagrin) [reply dwds/lib/src/debugging/debugger.dart:355]: Sure

Comment (annagrin) [reply dwds/lib/src/debugging/debugger.dart:357]: Done!

File: dwds/lib/src/debugging/debugger.dart [hunk 3]
```diff
@@ -338,14 +340,23 @@ class Debugger extends Domain {
   /// The variables visible in a frame in Dart protocol [BoundVariable] form.
   Future<List<BoundVariable>> variablesFor(
       List<dynamic> scopeChain, String callFrameId) async {
+    var isNativeObject = (dynamic v) {
```
[the remark this answers is not in the document]
Comment (grouma) [reply dwds/lib/src/debugging/debugger.dart:343]: Maybe even pull this out to a private helper

[the remark this answers is not in the document]
Comment (annagrin) [reply dwds/lib/src/debugging/debugger.dart]: done, thanks!

[the remark this answers is not in the document]
Comment (annagrin) [reply dwds/lib/src/debugging/debugger.dart]: done, thanks!

**Comment by annagrin** [comment]
> Should we add a test to `variable_scope_test.dart` or `chrome_proxy_service_test.dart` to verify the paused stack frames do not contain native javascript references?

makes sense, will add

**Comment by devoncarew** [comment]
@annagrin - when this lands, can you publish package:dwds and bring the new version into flutter? I'd like to verify the fix w/ IntelliJ; thanks!

**Comment by annagrin** [comment]
@devoncarew the matching flutter PR is merged: https://github.com/flutter/flutter/pull/56958
Please let me know if it works!

**Comment by devoncarew** [comment]
Looks great:

Screen Shot 2020-05-15 at 10 17 01 AM

Thanks!

**Pull request merged**